### PR TITLE
Andy/foreign key changes

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**
- Indexes can no longer be HIDDEN
- Foreign Keys must have a pre-existing index in the parent table
- Foreign Keys use pre-existing indexes in the child table if it exists

Still TODO: Diff, Merge, and Stage changes

### CODE REVIEW

File: bats/foreign-keys.bats [hunk 1]
```diff
@@ -597,759 +211,139 @@ SQL
...
-    ! [[ "$output" =~ "(parent_id) HIDDEN" ]] || false
+    [[ "$output" =~ "(v1)" ]] || false
     run dolt schema show child
     [ "$status" -eq "0" ]
     ! [[ "$output" =~ 'CONSTRAINT `fk_name`' ]] || false
     run dolt sql -q "ALTER TABLE child DROP FOREIGN KEY fk_name"
     [ "$status" -eq "1" ]
 }
 
-@test "foreign-keys: ALTER TABLE SET NULL on non-nullable column" {
```
Comment (Hydrocharged) [inline bats/foreign-keys.bats:1209]: Why remove this test?

File: go/libraries/doltcore/sqle/table_editor_fk_test.go [hunk 2]
```diff
@@ -384,7 +379,9 @@ func assertTableEditorRows(t *testing.T, root *doltdb.RootValue, expected []sql.
 			sqlRows = append(sqlRows, sqlRow)
 			return nil
 		})
-		assert.Equal(t, convertSqlRowToInt64(expected), sqlRows)
+		if !assert.Equal(t, convertSqlRowToInt64(expected), sqlRows) {
+			fmt.Println("")
```
Comment (Hydrocharged) [inline go/libraries/doltcore/sqle/table_editor_fk_test.go:383]: I'm assuming you wrote something here and forgot to remove it afterward lol

Comment (andrew-wm-arthur) [reply go/libraries/doltcore/sqle/table_editor_fk_test.go:383]: Yeah I needed somewhere to put a breakpoint, good catch

File: bats/foreign-keys.bats (hunk 1 above)
Comment (andrew-wm-arthur) [reply bats/foreign-keys.bats:1209]: good call

**Pull request closed without merging**